app/src/main/java/org/p2p/wallet/jupiter/repository/tokens/db/SwapTokensDao.kt

```diff
@@ -0,0 +1,49 @@
+package org.p2p.wallet.jupiter.repository.tokens.db
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.Query
+import org.intellij.lang.annotations.Language
+
+@Dao
+interface SwapTokensDao {
+    private companion object {
+        @Language("RoomSql")
+        private const val INDEX_FOR_ADDRESS =
+            "SELECT ordinal_index FROM swap_tokens WHERE address = :mintAddress"
+
+        @Language("RoomSql")
+        private const val SWAPPABLE_TOKEN_INDEXES_FOR_TOKEN =
+            "SELECT index_swappable_token FROM swap_tokens_routes WHERE index_source_token = ($INDEX_FOR_ADDRESS)"
+    }
+
+    @Insert
+    suspend fun insertSwapTokens(entities: List<SwapTokenEntity>)
+
+    @Insert
+    suspend fun insertTokenRoutes(routes: List<SwapTokenRouteCrossRef>)
+
+    @Query("SELECT * FROM swap_tokens WHERE ordinal_index IN ($SWAPPABLE_TOKEN_INDEXES_FOR_TOKEN)")
+    suspend fun getSwappableTokens(mintAddress: String): List<SwapTokenEntity>
+
+    @Query(
+        "SELECT * FROM swap_tokens " +
+            "WHERE ordinal_index IN ($SWAPPABLE_TOKEN_INDEXES_FOR_TOKEN) " +
+            "AND (symbol LIKE :mintAddressOrSymbol " +
+            "OR address LIKE :mintAddressOrSymbol)"
```

Comment:
maybe it's worth adding `LOWER(symbol) LIKE :mintAddressOrSymbol` and in the function call writing a note that it must be called with lowercased token symbol?` I'm afraid for queries like `BOnk`